lib/options.go

```diff
@@ -44,6 +47,238 @@ const DefaultScenarioName = "default"
 // nolint: gochecknoglobals
 var DefaultSummaryTrendStats = []string{"avg", "min", "med", "max", "p(90)", "p(95)"}
 
+// DNSConfig is the DNS resolver configuration.
+type DNSConfig struct {
+	// If positive, defines how long DNS lookups should be returned from the cache.
+	TTL null.String `json:"ttl"`
+	// Select specifies the strategy to use when picking a single IP if more than one is returned for a host name.
+	Select NullDNSSelect `json:"select"`
+	// Policy specifies how to handle returning of IPv4 or IPv6 addresses.
+	Policy NullDNSPolicy `json:"policy"`
+	// FIXME: Valid is unused and is only added to satisfy some logic in
+	// lib.Options.ForEachSpecified(), otherwise it would panic with
+	// `reflect: call of reflect.Value.Bool on zero Value`.
+	Valid bool
+}
+
+// DNSPolicy specifies the preference for handling IP versions in DNS resolutions.
+//go:generate enumer -type=DNSPolicy -trimprefix DNS -output dns_policy_gen.go
+type DNSPolicy uint8
+
+// These are lower camel cased since enumer doesn't support it as a transform option.
+// See https://github.com/alvaroloes/enumer/pull/60 .
+const (
+	// DNSpreferIPv4 returns an IPv4 address if available, falling back to IPv6 otherwise.
+	DNSpreferIPv4 DNSPolicy = iota + 1
+	// DNSpreferIPv6 returns an IPv6 address if available, falling back to IPv4 otherwise.
+	DNSpreferIPv6
+	// DNSonlyIPv4 only returns an IPv4 address and the resolution will fail if no IPv4 address is found.
+	DNSonlyIPv4
+	// DNSonlyIPv6 only returns an IPv6 address and the resolution will fail if no IPv6 address is found.
+	DNSonlyIPv6
+	// DNSany returns any resolved address regardless of version.
+	DNSany
+)
+
+// UnmarshalJSON converts JSON data to a valid DNSPolicy
+func (d *DNSPolicy) UnmarshalJSON(data []byte) error {
+	if bytes.Equal(data, []byte(`null`)) {
+		return nil
+	}
+	var s string
+	if err := json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+	v, err := DNSPolicyString(s)
+	if err != nil {
+		return err
+	}
+	*d = v
+	return nil
+}
+
+// MarshalJSON returns the JSON representation of d
+func (d DNSPolicy) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+// NullDNSPolicy is a nullable wrapper around DNSPolicy, required for the
+// current configuration system.
+type NullDNSPolicy struct {
```

Comment:
Should we move some of these DNS config types to `lib/types`, like Mihail did in 46d7dd0c?

Reply:
Ideally ... but I would argue this can be done in a second PR or at least as the last step, but if you have time - go for it

Reply:
Resolved by caf82ab9.